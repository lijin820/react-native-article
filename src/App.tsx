import 'react-native-gesture-handler';
import React from 'react';
import { Provider } from 'react-redux';
import { NavigationContainer } from '@react-navigation/native';

import configureStore from './redux/store';
import AppNavigator from './screens/AppNavigator';

const store = configureStore();

export default function App() {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <AppNavigator />
      </NavigationContainer>
    </Provider>
  );
}
