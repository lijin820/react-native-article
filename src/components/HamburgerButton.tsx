import React from 'react';
import { TouchableOpacity, StyleSheet } from 'react-native';
import Ionicons from 'react-native-vector-icons/Ionicons';

export function HamburgerButton() {
  return (
    <TouchableOpacity
      onPress={() => console.log('clicked')}
      style={styles.burgerBtn}>
      <Ionicons name={'menu-outline'} size={25} color={'black'} />
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  burgerBtn: {
    marginLeft: 10,
  },
});
