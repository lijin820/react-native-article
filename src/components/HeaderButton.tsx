import React, { Fragment } from 'react';
import { TouchableOpacity, StyleSheet, View } from 'react-native';
import Ionicons from 'react-native-vector-icons/Ionicons';

type HeaderButtonProps = {
  align?: string;
};

export function HeaderButton({ align }: HeaderButtonProps) {
  return (
    <Fragment>
      {align === 'right' ? (
        <View style={styles.rightBtnWrapper}>
          <TouchableOpacity onPress={() => console.log('clicked')}>
            <Ionicons
              name={'search-outline'}
              size={25}
              color={'black'}
              style={styles.rightIcon}
            />
          </TouchableOpacity>
          <TouchableOpacity onPress={() => console.log('clicked')}>
            <Ionicons
              name={'notifications-outline'}
              size={25}
              color={'black'}
              style={styles.rightIcon}
            />
          </TouchableOpacity>
        </View>
      ) : (
        <TouchableOpacity
          onPress={() => console.log('clicked')}
          style={styles.burgerBtn}>
          <Ionicons name={'menu-outline'} size={25} color={'black'} />
        </TouchableOpacity>
      )}
    </Fragment>
  );
}

const styles = StyleSheet.create({
  burgerBtn: {
    marginLeft: 10,
  },
  rightBtnWrapper: {
    flexDirection: 'row',
  },
  rightIcon: {
    marginRight: 10,
  },
});
