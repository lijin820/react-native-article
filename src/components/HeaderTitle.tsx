import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

type HeaderProps = {
  title: string;
};

export function HeaderTitle(props: HeaderProps) {
  const { title } = props;
  return <Text style={styles.title}>{title}</Text>;
}

const styles = StyleSheet.create({
  title: {
    fontSize: 20,
    fontFamily: 'Raleway-Medium',
  },
});
