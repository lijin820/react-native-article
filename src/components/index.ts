export * from './HeaderTitle';
export * from './LoadingSpinner';
export * from './HamburgerButton';
