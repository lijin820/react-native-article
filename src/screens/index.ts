export * from './Latest';
export * from './Learning';
export * from './Discussions';
export * from './Search';
export * from './Resources';
