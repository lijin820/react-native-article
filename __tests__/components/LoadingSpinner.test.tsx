import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { SafeAreaView } from 'react-native';
import { LoadingSpinner } from '../../src/components';

test('renders the Loading Spinner', () => {
  const component = shallow(<LoadingSpinner />);
  expect(component).toMatchSnapshot();
});
