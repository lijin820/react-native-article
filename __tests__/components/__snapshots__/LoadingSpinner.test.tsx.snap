// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Loading Spinner 1`] = `ShallowWrapper {}`;
